test/Microsoft.Framework.PackageManager.FunctionalTests/DnuSourcesTests.cs

```diff
@@ -0,0 +1,99 @@
+namespace Microsoft.Framework.PackageManager
+{
+    [Collection(nameof(PackageManagerFunctionalTestCollection))]
+    public class DnuSourcesTests
+    {
+        private readonly PackageManagerFunctionalTestFixture _fixture;
+
+        public DnuSourcesTests(PackageManagerFunctionalTestFixture fixture)
+        {
+            _fixture = fixture;
+        }
+
+        public static IEnumerable<object[]> RuntimeComponents
+        {
+            get
+            {
+                return TestUtils.GetRuntimeComponentsCombinations();
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(RuntimeComponents))]
+        public void DnuSources_ListsAllSources(string flavor, string os, string architecture)
+        {
+            bool isWindows = TestUtils.CurrentRuntimeEnvironment.OperatingSystem == "Windows";
+            var environment = new Dictionary<string, string>
+            {
+                { "DNX_TRACE", "0" },
+            };
+
+            var rootConfig =
+@"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+  <packageSources>
+    <add key=""Source1"" value=""https://source1"" />
+    <add key=""Source2"" value=""https://source2"" />
+  </packageSources>
+</configuration>";
+
+            var subConfig =
+@"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+  <packageSources>
+    <add key=""Source3"" value=""https://source3"" />
+  </packageSources>
+  <disabledPackageSources>
+    <add key=""Source1"" value=""https://source1"" />
+  </disabledPackageSources>
+</configuration>";
+
+            var projectStructure =
+@"{
+    'root': {
+        'NuGet.config': """",
+        'sub': {
+            'NuGet.config': """"
+        }
+    }
+}";
+
+            var expectedContent = @"1.  Source1                       https://source1 [Disabled]
+2.  Source2                       https://source2
+3.  Source3                       https://source3";
+            var runtimeHomePath = _fixture.GetRuntimeHomeDir(flavor, os, architecture);
+            using (var testEnv = new DnuTestEnvironment(runtimeHomePath, projectName: "Project Name"))
+            {
+                var projectPath = testEnv.ProjectPath;
+                DirTree.CreateFromJson(projectStructure)
+                    .WithFileContents("root/NuGet.config", rootConfig)
+                    .WithFileContents("root/sub/NuGet.config", subConfig)
+                    .WriteTo(projectPath);
+
+                string output;
+                string error;
+                var exitCode = DnuTestUtils.ExecDnu(
+                    runtimeHomePath,
+                    subcommand: "sources",
+                    arguments: "root/sub",
+                    stdOut: out output,
+                    stdErr: out error,
+                    environment: environment,
+                    workingDir: projectPath);
+
+                Assert.Equal(0, exitCode);
+                Assert.Empty(error);
+                Assert.Contains(expectedContent, output);
```

Comment:
Yeah, I'm trying to find a balance between a brittle test and testing the actual functionality. I think I can put the whole of the expected output here, but from experience that can lead to a lot of tests breaking from a small change :). Here, I think it'll be fine, at least until we deal with localization :P